File: main/src/main/java/com/example/main/ui/MainViewModel.java
package com.example.main.ui;

import android.arch.lifecycle.ViewModel;

import com.example.core.data.model.Resource;
import com.example.core.data.model.movie.Movie;
import com.example.core.util.SingleLiveEvent;
import com.example.main.data.source.remote.MainRepository;

public class MainViewModel extends ViewModel {

    SingleLiveEvent<Resource<Movie>> movieSingleLiveEvent = new SingleLiveEvent<>();
    private MainRepository mainRepository = new MainRepository();

    public void getListMovies(Integer page) {
        mainRepository.getMovies(movieSingleLiveEvent, page);
    }

    public void getListMoviesTop(Integer page) {
        mainRepository.getTopMovies(movieSingleLiveEvent, page);
    }
}

File: main/src/main/java/com/example/main/ui/MainViewModel.kt
package com.example.main.ui

import android.arch.lifecycle.ViewModel

import com.example.core.data.model.Resource
import com.example.core.data.model.movie.Movie
import com.example.core.util.SingleLiveEvent
import com.example.main.data.source.remote.MainRepository

class MainViewModel : ViewModel() {

    var movieSingleLiveEvent = SingleLiveEvent<Resource<Movie>>()
    private val mainRepository = MainRepository()

    fun getListMovies(page: Int) {
        mainRepository.getMovies(movieSingleLiveEvent, page)
    }

    fun getListMoviesTop(page: Int) {
        mainRepository.getTopMovies(movieSingleLiveEvent, page)
    }
}
